src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/SchedulerTimestampRepositoryExtensionsImpl.java

```diff
@@ -0,0 +1,35 @@
+package uk.nhs.digital.nhsconnect.nhais.repository;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+import java.time.LocalDateTime;
+
+import lombok.RequiredArgsConstructor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SchedulerTimestampRepositoryExtensionsImpl implements SchedulerTimestampRepositoryExtensions {
+
+    private final MongoOperations mongoOperations;
+
+    @Override
+    public boolean updateTimestamp(UpdateTimestampParams updateTimestampParams, UpdateTimestampDetails updateTimestampDetails) {
```

Comment:
Strings could be extracted to static fields

Reply:
Maybe. We are using similar approach in other places. Bad thing about java that it's missing `nameof()` built-in function ...